### ISSUE TITLE
Unicode SOABI-related changes

### ISSUE THREAD

**Issue by njsmith** [pypa/manylinux#29]
Given discussion [here](https://mail.python.org/pipermail/distutils-sig/2016-February/028251.html) and [here](https://mail.python.org/pipermail/distutils-sig/2016-February/028252.html), it sounds like we can/should drop the UCS4-only requirement from PEP 513 after all.

If that's right, then I guess there are a few things to do:
- [x] Update PEP text
- [x] Update pip pull request
- [x] Update our `check-manylinux.py` script
- [ ] Update auditwheel to remove the UCS2 check (but possibly it should unconditionally require the use of an SOABI tag, just to nudge people into doing the right thing?)
- [x] Update the docker image to include both UCS2 and UCS4 builds for python 2.6 / 2.7. (Also, make sure that it includes a recent-enough wheel package.)

For the docker images, I propose the following layout:

```
/opt/2.7.11-ucs2
/opt/2.7.11-ucs4
/opt/2.7-ucs2 -> 2.7.11-ucs2
/opt/2.7-ucs4 -> 2.7.11-ucs4
/opt/2.7 -> 2.7-ucs4
```

**Comment by rmcgibbo** [comment]
On a slightly separate but related note, I wonder if we should rearrange the Pythons layout in a way such that a build script like
```
for $py in /opt/or/some/other/path/*/bin/python; do
   echo $py
   ...
done
```
would work nicely and only produce 1 wheel for each unique Python.

**Comment by njsmith** [comment]
...yeah, that would be nice, wouldn't it :-)

Option 1: just install into `/opt/3.3`, `/opt/3.4`, etc., and skip the whole symlink thing entirely.

Option 2: put symlinks in `/python/3.2`, `/python/3.3`, etc.

Option 3: ...?

**Comment by rmcgibbo** [comment]
On distutils-sig, it looks like Nick"s suggestion was to change the PEP text, but leave the docker images as-is.

And we also might need to change the pip PR too?

**Comment by njsmith** [comment]
> On distutils-sig, it looks like Nick"s suggestion was to change the PEP text, but leave the docker images as-is.

Yeah... I don't have a strong opinion either way. On the one hand, UCS-2 builds are indeed broken, and probably rare (though this isn't really clear). On the other hand, it's kinda up to individual projects whether they decide to provide UCS-2 wheels, and if they want to use the docker image to do so then it's no skin off my nose. Projects that don't want to provide such builds can always skip that regardless.

> And we also might need to change the pip PR too?

Good point, edited the checklist above to reflect this.

**Comment by njsmith** [comment]
changes to PEP and `check-manylinux1.py` are in #31

**Comment by njsmith** [comment]
Now that #35 is merged, it looks like all that's left here is fixing auditwheel (pypa/auditwheel#16)
